### ISSUE TITLE
Waitlisted - ReLice

### ISSUE THREAD

**Issue by KumarManas04** [LNMHacks/submissions#8]
**Before you start, please follow this format for your issue title**:  
Waitlisted - ReLice

## ℹ️ Project information
_Please complete all applicable._

- **Project Name**: ReLice
- **Short Project Description**: Help find missing children through machine learning and power of masses
- **Team Name**: Waitlisted
- **Team Members**: 
Kumar Manas (KumarManas04)
Anubhav Natani(ANUBHAVNATANI)
Sourabh Tripathi(Sourabhtripathi)
Akshat Jain(iamakshatjain)
- **Where to find you?**: LT-19
- **Demo Link**: 
Presentation - https://docs.google.com/presentation/d/1vExc2VuGjImSqGrL_lvfmN1y8Yfo7wCZ_ZQVhyJ-Ehg/edit?usp=sharing
Website - https://relice-test.azurewebsites.net/
App - https://github.com/KumarManas04/ReLice/blob/master/ReLice.apk
***Demo videos for -*** 
Android app -https://github.com/KumarManas04/ReLice/blob/master/Waitlisted_Demo_Videos/Android_App_Demo_Video.mp4
React Website -
https://github.com/KumarManas04/ReLice/blob/master/Waitlisted_Demo_Videos/ReactVideo.mp4
ML_API -
https://github.com/KumarManas04/ReLice/blob/master/Waitlisted_Demo_Videos/ML_API_Demo_Video.mp4
API-https://github.com/KumarManas04/ReLice/blob/master/Waitlisted_Demo_Videos/API_Demo_Video.webm
 
- **Repository Link(s)**: https://github.com/KumarManas04/ReLice
- **Prize Categories Enrolled for**: 
   (Please tick any specific Category that applies)
  - [x] Best use of Microsoft Azure
  - [ ] Best Beginner (Your team should be atleast 50% Beginners)
  - [x] Best "Jugaad" - Best Fun/ Silly/ Innovative Hack
  - [ ] Best UIPath Hack

## 🔥 Your Pitch
63407 children went missing in 2016, according to data from the National Crime Records Bureau, which translates to an average of 174 every day. In Delhi alone, 27,356 children went missing during 2015-2018, and more disquietingly, 50% of the children who have gone missing until 2016 have stayed untraced. This is a big issue, and most of these children get caught up in trafficking or child labor.

For solving this problem to some extent, we have developed ReLice. A software solution for the issue of missing children. The major problem is that there is a lot of workload on the authorities. They can’t be at every place. But people are present almost everywhere. So the main aim of our project is to transfer this load on the authorities to the masses. We are using machine learning and computer vision algorithms, along with a lot of Azure services. The users across the country can send images of suspected missing children, which we compare to the database of reported missing children. If a match is found, authorities are immediately informed so that they can search the area for the missing children. 

We have the following components in our software-
ML Flask App (Azure App Service, Azure Face API)
Database API (Permanent DB, Temporary DB, Matched DB) Azure Cosmos DB
Android App (Azure Cognitive Service)
React App (Azure App Service)

We know that there can be a lot of edge case in this software here are some which we have tackled. 

1. The project maintains a temporary database where queries are stored for 72 hours. It deals with the case in which a child’s photo is submitted for a check before the child has been reported missing by the parents. When a child is reported missing, his images are compared with the temporary database.

2. We have secured the data of the children by not storing the images of the child at all. We submit the image to the Face API of Azure, and it generates a face id. Now the image can be accessed by our APIs only and not by anyone else. Even our API accesses the pictures for comparison via the face ids and not the actual images. This ensures total security of everyone’s data.

We have used many technologies to build this app, some of which include Microsoft Azure, Python, Java, JavaScript, Kotlin, Android, React, Pytorch, Node JS, Keras.

## 🔦 Any other specific thing you want to highlight?
Our project protects the privacy of the individuals whose images are uploaded. We have achieved this by-
- Not storing actual images. Instead the images are provided to Azure Face API and it provides us with a face id which can be used for all purposes of our project. So the images remain safe with Azure which keeps these images private.
- Immediately delete the data of the children that have been found and securing the access portal which is present on the side of the authorities.
- [System Design](https://imgur.com/gGthhrf)

## ✅ Checklist

**Before you post the issue**:
- [x] You have followed the issue title format.
- [x] You have mentioned the correct labels.
- [x] You have provided all the information correctly.
